pkg/chartutil/jsonschema.go

```diff
@@ -91,3 +99,36 @@ func ValidateAgainstSingleSchema(values Values, schemaJSON []byte) (reterr error
 
 	return nil
 }
+
+func validateUsingNewValidator(valuesJSON, schemaJSON []byte) error {
+	schema, err := jsonschema.UnmarshalJSON(bytes.NewReader(schemaJSON))
+	if err != nil {
+		return err
+	}
+	values, err := jsonschema.UnmarshalJSON(bytes.NewReader(valuesJSON))
+	if err != nil {
+		return err
+	}
+
+	compiler := jsonschema.NewCompiler()
+	err = compiler.AddResource("file:///values.schema.json", schema)
+	if err != nil {
+		return err
+	}
+
+	validator, err := compiler.Compile("file:///values.schema.json")
+	if err != nil {
+		return err
+	}
+
+	return validator.Validate(values)
+}
+
+func schemaIs2020(schemaJSON []byte) bool {
+	var partialSchema struct {
+		Schema string `json:"$schema"`
+	}
+	_ = json.Unmarshal(schemaJSON, &partialSchema)
+	url, _ := url.Parse(partialSchema.Schema)
```

Comment:
We should not elide the error here.

Reply:
that said, iiuc, the only correct value for `$schema` is `https://json-schema.org/draft/2020-12/schema`

So IMHO a simple is correct / enough:
```suggestion
	return "https://json-schema.org/draft/2020-12/schema" == partialSchema.Schema
```

this also means if any other other portions of the URL are specified e.g. query params, the schema version string will (rightly and pedantically) be rejected

Reply:
TBH, I don't want to be pedantic here because I've seen people using `http://json-schema.org/draft/2020-12/schema#` in the wild

Reply:
wdyt?, I don't see any harm in relaxing this thing

Reply:
> TBH, I don't want to be pedantic here because I've seen people using https://json-schema.org/draft/2020-12/schema#/ in the wild

They should IMHO use the correct value :) We shouldn't be in the business of trying to "fix" invalid values for the user. 

I think in Helm's case, they would introduce a wrong value into their `values.schema.json`, get an error, then fix the wrong value. Ideally the error will show the allowed values (which would be an improvement for this PR).

We should also validate that the path, query, fragment & user are empty. And scheme is `https`.

Reply:
updated, following your suggestion